packages/ui/src/views/secrets/secret-entity-form.tsx

```diff
@@ -0,0 +1,107 @@
+import { useMemo } from 'react'
+
+import { EntityIntent, TranslationStore } from '@/views'
+import { Alert } from '@components/alert'
+import { Button } from '@components/button'
+import { EntityFormLayout } from '@views/unified-pipeline-studio/components/entity-form/entity-form-layout'
+import { EntityFormSectionLayout } from '@views/unified-pipeline-studio/components/entity-form/entity-form-section-layout'
+
+import { InputFactory, RenderForm, RootForm, useZodValidationResolver } from '@harnessio/forms'
+
+import { AnySecretDefinition, onSubmitSecretProps } from './types'
+
+interface SecretEntityFormProps {
+  onFormSubmit?: (values: onSubmitSecretProps) => void
+  getSecretDefinition?: () => AnySecretDefinition
+  onBack?: () => void
+  useTranslationStore: () => TranslationStore
+  inputComponentFactory: InputFactory
+  apiError?: string | null
+  intent: EntityIntent
+}
+
+export const SecretEntityForm = (props: SecretEntityFormProps): JSX.Element => {
+  const {
+    apiError = null,
+    onFormSubmit,
+    getSecretDefinition,
+    onBack,
+    useTranslationStore,
+    inputComponentFactory,
+    intent
+  } = props
+  const { t: _t } = useTranslationStore()
+
+  const onSubmit = (data: onSubmitSecretProps) => {
+    onFormSubmit?.(data)
+  }
+
+  const formDefinition = useMemo(() => {
```

Comment:
why do we need this complex mapping function? formdef should work directly from config to the component.